include/gsl/string_span

```diff
@@ -108,18 +108,20 @@ namespace details
 // Will fail-fast if sentinel cannot be found before max elements are examined.
 //
 template <typename T, const T Sentinel>
-span<T, dynamic_extent> ensure_sentinel(T* seq, std::ptrdiff_t max = PTRDIFF_MAX)
+span<T, dynamic_extent> ensure_sentinel(T* seq,
+                                        std::size_t max = std::numeric_limits<std::size_t>::max())
```

Comment:
Pre-existing: I don't see `<limits>` included for `numeric_limits`, and this call to `max` will be stomped over by `<Windows.h>` macros. I recommend using `static_cast<std::size_t>(-1)` here. Ditto below.